ci/docker-compose.yml

```diff
@@ -0,0 +1,75 @@
+version: '3'
+services:
+
+  postgres:
+    image: postgres
+    ports:
+      - 5432:5432
+    environment:
+      POSTGRES_PASSWORD: postgres
+
+  impala:
+    image: ibisproject/impala
+    hostname: impala
+    environment:
+      PGPASSWORD: postgres
+    ports:
+      # HDFS
+      - 9020:9020
+      - 50070:50070
+      - 50075:50075
+      - 8020:8020
+      - 8042:8042
+      # Hive
+      - 9083:9083
+      # Impala
+      - 21000:21000
+      - 21050:21050
+      - 25000:25000
+      - 25010:25010
+      - 25020:25020
+
+  clickhouse:
+    image: yandex/clickhouse-server:latest
+    ports:
+      - 8123:8123
+      - 9000:9000
+
+  waiter:
+    image: jwilder/dockerize
+    command: |
+      dockerize -wait tcp://postgres:5432
+                -wait tcp://impala:21050
+                -wait tcp://impala:50070
+                -wait tcp://clickhouse:9000
+                -wait-retry-interval 5s
+                -timeout 5m
+
+  ibis:
+    image: ibis:${PYTHON_VERSION-3.6}
+    environment:
+      - IBIS_TEST_DOWNLOAD_DIRECTORY=/tmp
+      - IBIS_TEST_DATA_DIRECTORY=/tmp/ibis-testing-data
+      - IBIS_TEST_SQLITE_DATABASE=/tmp/ibis_testing.db
+      - IBIS_TEST_NN_HOST=impala
+      - IBIS_TEST_IMPALA_HOST=impala
+      - IBIS_TEST_IMPALA_PORT=21050
+      - IBIS_TEST_WEBHDFS_PORT=50070
+      - IBIS_TEST_WEBHDFS_USER=hdfs
+      - IBIS_TEST_POSTGRES_HOST=postgres
+      - IBIS_TEST_POSTGRES_PORT=5432
+      - IBIS_TEST_POSTGRES_USER=postgres
+      - IBIS_TEST_POSTGRES_PASSWORD=postgres
+      - IBIS_TEST_POSTGRES_DATABASE=ibis_testing
+      - IBIS_TEST_CLICKHOUSE_HOST=clickhouse
+      - IBIS_TEST_CLICKHOUSE_PORT=9000
+      - IBIS_TEST_CLICKHOUSE_DATABASE=ibis_testing
+      - GOOGLE_BIGQUERY_PROJECT_ID=ibis-gbq
+      - GOOGLE_APPLICATION_CREDENTIALS=/tmp/gcloud-service-key.json
+    volumes:
+      - /tmp/ibis:/tmp
+    build:
+      context: ..
+      dockerfile: ci/Dockerfile
+      args:
+        PYTHON: ${PYTHON_VERSION-3.6}
```

Comment:
Let's use the `${parameter:-word}` syntax so that this will substitute `word` if `parameter` is `null`.
![shell](https://user-images.githubusercontent.com/417981/35631704-8d31db58-0672-11e8-958f-c02fedf1a0f3.png)